PaloAltoNetworks/terraform-aws-vmseries-modules
# feat(modules/bootstrap): Allow the user to bring their own role/policy

## Description

The current bootstrap module creates an IAM policy/role by default when using the bootstrap module.  PR delivers refactored ``boostrap`` module to allow the user to bring their own role/policy.

Moreover PR was extended by adding missing DHCP variables:
- `dhcp_send_hostname`
- `dhcp_send_client_id`
- `dhcp_accept_server_hostname`
- `dhcp_accept_server_domain`

## Motivation and Context

PR resolves issue #217 and extends module bootstrap about 4 variables DHCP (details in PR's comments).

## How Has This Been Tested?

Integration tests, which are using example code, were prepared in ``tests/bootstrap`` folder. In ``tests/bootstrap/README.md`` there is short instruction how to run them on local machine. There were prepared 3 tests for checking 3 behaviours for ``bootstrap`` module:
* verify bootstrap module output, while using default approach - creating IAM role with policy
* verify error while running ``terraform plan`` for bootstrap module, while variable ``create_iam_role_policy`` was set to ``false`` and no name of IAM role was provided
* verify bootstrap module output, while using variable ``create_iam_role_policy`` was set to ``false`` and providing IAM role name in variable ``iam_role_name``

Besides preparing ``terratest``, changes were checked using currently available examples:
* ``tgw_inbound_combined_with_gwlb``
* ``standalone_vmseries_with_userdata_bootstrap``

## Screenshots (if appropriate)

Not appropriate.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist
